### ISSUE TITLE
Need easy way to get generation in which Pokemon form/variant was introduced from Pokemon endpoint

### ISSUE THREAD

**Issue by GregLyons** [PokeAPI/pokeapi#656]
Hello,

I'm currently working with PokeAPI, and I'm having trouble determining the generation of a given Pokemon that I've queried from a Pokemon endpoint (NOT a Pokemon Species endpoint). For the reasons I describe below, I believe that the Pokemon endpoint should also list the generation in which a Pokemon form/variant first appears, as the Pokemon Species endpoint is not capable of this, and there's no easy way to get this data from information already contained in the Pokemon endpoint.

From the documentation, when you query the Pokemon endpoint, e.g. `.../pokemon/bulbasaur` (`...` will refer to `https://pokeapi.co/api/v2/`) you get things like moves, abilities, etc. However, I can't find a field which gives the generation in which that Pokemon appears. For example, let's say I want to figure out which generation Mega Charizard X appears in. I can't just use `.../pokemon-species/charizard-mega-x` and get the `generation` field there, since that page doesn't even exist. In general, variants and forms of Pokemon aren't accessible via `.../pokemon-species`. 

For most Pokemon, the form/variant is present in the same generation as the original Pokemon (e.g. the forms of Wormadam), or, in the case of Mega Evolutions and GMaxing, in the generation of that mechanic. Thus, perhaps one could just see whether the name has "Mega" in it, or they could see what the species of the variant is (e.g. Wormadam Trash is of the species Wormadam). However, this isn't completely satisfactory to me because 
1. there are many exceptions (e.g. `eternatus-eternamax`) that I could miss.
2. if Game Freak ever decides to randomly introduce an alternate Pokemon form in a later game, say, they give Metapod a "red" variant in Generation 9 (exciting!), then one would have to track down every case of this in each future generation.

Also, the Pokemon's ID won't work, since many form/variants have IDs greater than 10,000.

My original thought was to write some code to figure it out based on the presence of the Pokemon's sprite/model (Gen <= 5 and Gen >= 6, respectively). The generation in which the Pokemon form/variant was introduced should be the first generation in which a sprite/model for that form/variant appears, right? You can access this in `.../pokemon/{pokemon name}.sprites.versions` 

Unfortunately, it seems that there exists Gen 5 sprites for Pokemon forms/variants introduced in later generations. Here are some examples I found while testing that my code works for Pokemon form/variants: 

1. Lycanroc Dusk has both a front and a back sprite in Gen 5: `.../pokemon/lycanroc-dusk.sprites.versions.generation-v.black-white.front_default` and `.../pokemon/lycanroc-dusk.sprites.versions.generation-v.black-white.back_default` both exist.
2. Mega Charizard X has a front sprite but not a back sprite in Gen 5: `.../pokemon/charizard-mega-x.sprites.versions.generation-v.black-white.front_default` exists but not `.../pokemon/charizard-mega-x.sprites.versions.generation-v.black-white.back_default`.
3. Deoxys-Speed has a front sprite but not a back sprite in Gen 3: `.../pokemon/deoxys-speed.sprites.versions.generation-iii.emerald.front_default`, but `.../pokemon/deoxys-speed.sprites.generation-iii.emerald.back_default`--instead, Deoxys Speed's Gen 3 back sprite is located in `.../pokemon/deoxys-speed.sprites.back_default`, outside its `versions` in its `sprites` property.

Let's say I tried to check for the generation in which a form/variant which had both front _and_ back sprites. Then Lycanroc would be a Gen 5 Pokemon, while Deoxys-Speed would be a Gen 4 Pokemon, and Mega Charizard X would make a cameo in Gen 5, but you can't have him on your team (since he has no back sprite).

To be clear, I don't mean the previous couple paragraphs as a critique of the way sprite data is stored in PokeAPI. Instead, I bring up these cases since going based off the first appearance of a form/variant's sprite/model was the only way I could think of to determine its original generation. I apologize in advance if there's an obvious way of doing this with the current API. 

EDIT: I just found [this .csv file in the pokedex API](https://github.com/PokeAPI/pokedex/blob/master-pokeapi/pokedex/data/csv/pokemon_form_generations.csv). There it assigns (possibly multiple) `generation_id` to a given `pokemon_form_id`. For example, Bulbasaur (`pokemon_form_id` 1) has `generation_id`'s 1, 2, 3, 4, 5, 6, and 7. So it seems like it only goes up to Gen 7, but that's pretty helpful. 

I'm still a bit concerned using this though, as either it has errors or I don't fully understand it. For example, `pokemon_form_id` 10001, Deoxys-Attack, not only has a `generation_id` of 3, 4, 5, 6, 7, but also of 2. Also, `pokemon_form_id` 10034, Mega Charizard X, has `generation_id`'s of 4 and 5 in addition to 6 and 7. Thus, I can't just take a `pokemon_form_id` and look at its minimum `generation_id` as I was hoping to.

**Comment by C-Garza** [comment]
As you've noticed the `/pokemon-species` does have a `generation` property that tells you when that pokemon was introduced. From what I see, you're having troubled seeing when the forms were introduced. 

If that's the case, you might want to check out the endpoint `/pokemon-form` and the `version_group` property. That tells you the version group that form was introduced in. You can use that to check which generation they were introduced in.
Here is the `/pokemon-form` endpoint for Charizard-Mega-X: [Charizard-Mexa-X](https://pokeapi.co/api/v2/pokemon-form/charizard-mega-x)

**Comment by GregLyons** [comment]
Ah, thanks @C-Garza! As I had feared, I should've read the documentation more carefully.

Yesterday I ended up doing the name-parser thing that I mentioned wanting to avoid doing. After looking [at this page on Bulbapedia](https://bulbapedia.bulbagarden.net/wiki/List_of_Pok%C3%A9mon_with_form_differences), I _believe_ indeed that all the alternative forms of a Pokemon species are either of the same generation as their base/default form (except Pikachu with the hats in Gen 7), then there are Mega/Primals where were all introduced in Gen 6, the regional forms in Gens 7 and 8, and finally the max farms are of course Gen 8. With that info, it's not so hard to determine the generation based on the name as I had originally thought. 

(Not sure if I should "Close with comment" this issue since I'm new to GitHub, but I think the issue is solved!)
